django/core/management/base.py

```diff
@@ -71,6 +72,18 @@ def error(self, message):
         else:
             raise CommandError("Error: %s" % message)
 
+    def add_subparsers(self, **kwargs):
+        parser_class = kwargs.get("parser_class", type(self))
+        if issubclass(parser_class, CommandParser):
+            kwargs.setdefault(
```

Comment:
Thanks for progressing this @adamchainz 👍 

I'm unsure that calling `setdefault` is the desired behaviour here.  I'm thinking that the wrapped class should be used in every case that hits this line, even if `CommandParser` or a custom subclass was passed in by the caller.

Reply:
That's right, fix pushed!